### PR TITLE
Integrating recid with latest (master).

### PR DESCRIPTION
**Pull request by gbenattar**
T: cargo test

### CODE REVIEW

File: src/protocols/two_party_ecdsa/lindell_2017/party_one.rs [hunk 1]
```diff
@@ -493,7 +495,19 @@ impl Signature {
         let s_tag_tag = BigInt::mod_mul(&k1_inv, &s_tag.0, &FE::q());
         let s = cmp::min(s_tag_tag.clone(), FE::q().clone() - s_tag_tag.clone());
 
-        Signature { s, r: rx }
+        /*
+         Calculate recovery id - it is not possible to compute the public key out of the signature
+         itself. Recovery id is used to enable extracting the public key uniquely.
+         1. id = R.y & 1
+         2. if (s > curve.q) id = id ^ 1
```
Comment (omershlo) [inline src/protocols/two_party_ecdsa/lindell_2017/party_one.rs]: should be  `curve.q /2 `

File: src/protocols/two_party_ecdsa/lindell_2017/party_one.rs [hunk 2]
```diff
@@ -493,7 +495,19 @@ impl Signature {
...
+        */
+        let is_ry_odd = ry.tstbit(0);
+        let mut recid = if is_ry_odd { 1 } else { 0 };
+        if s_tag_tag.gt(&FE::q().div_floor(&BigInt::from(2))) {
```
Comment (omershlo) [inline src/protocols/two_party_ecdsa/lindell_2017/party_one.rs]: not the safest way to write it. 
It is better to write:
```
s_1 = s_tag_tag;
s_2 = FE::q() - s_tag_tag;
if s_1 > s_2 {recid = recid ^ 1;}
```

File: src/protocols/two_party_ecdsa/lindell_2017/party_one.rs [hunk 3]
```diff
@@ -481,6 +482,7 @@ impl Signature {
         r = r.scalar_mul(&ephemeral_local_share.secret_share.get_element());
 
         let rx = r.x_coor().unwrap().mod_floor(&FE::q());
+        let ry = r.y_coor().unwrap().mod_floor(&FE::q());
```
Comment (omershlo) [inline src/protocols/two_party_ecdsa/lindell_2017/party_one.rs]: I'm afraid that this trick will not work with other curves. Can you test it when using curv with different elliptic curves? ones where `r.y_coor()` is defined.  If this trick cannot be generalize than I suggest to either move it to KMS / to have to signing functions, one with recid_support and a regular one without

**Comment by gbenattar** [comment]
@oleiba can you look into it?

File: src/protocols/two_party_ecdsa/lindell_2017/party_one.rs (hunk 1 above)
Comment (gbenattar) [reply src/protocols/two_party_ecdsa/lindell_2017/party_one.rs]: Can you explain? I think @oleiba tested with this code signing for ETH.

Comment (omershlo) [reply src/protocols/two_party_ecdsa/lindell_2017/party_one.rs]: I meant the comment. the code is fine

**Pull request merged**